packages/headless/src/features/search/search-actions-loader.ts

```diff
@@ -0,0 +1,55 @@
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+import {AsyncThunkSearchOptions} from '../../api/search/search-api-client';
+import {Engine} from '../../app/engine';
+import {search} from '../../app/reducers';
+import {SearchAction} from '../analytics/analytics-utils';
+import {
+  executeSearch,
+  ExecuteSearchThunkReturn,
+  fetchMoreResults,
+  StateNeededByExecuteSearch,
+} from './search-actions';
+
+/**
+ * The search actions.
+ */
+export interface ISearchActions {
```

Comment:
Just a question about the `IFoo` pattern for interface: It's not something we use at all in other places in Headless, so why start here ?

Reply:
It's a practice we had for JSUI but it would go against our styleguide here

Reply:
The `SearchActions` name is currently used by the namespace grouping the actions. Until it is deprecated and removed, I needed a different name for this interface to be able to correctly extract and generate the documentation. The used the `I` prefix more to avoid a collision, not to introduce a new convention per say.

Some other possibilities could be `AllSearchActions`, `LoadedSearchActions`, `SearchActionCreators`.

```
 loadSearchActions(engine: Engine<object>): AllSearchActions

 loadSearchActions(engine: Engine<object>): LoadedSearchActions

 loadSearchActionCreators(engine: Engine<object>): SearchActionCreators
```

While the last one is the most correct, it's also the most verbose,

Any preferences? @kaitlynkatz, tagging you too since the interface name will show up in the documentation, unless they get post-processed.

Reply:
"Verbose" is not necessarily bad... 

It does convey more info about "what this function actually does". 

As long as it does not go overboard...
I'm fine with `SearchActionCreators`

Reply:
+1 for `SearchActionCreators`, the `I` is still a convention & it could lead to more of its usage. The [Typescript repo own guidelines don't allow for it](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names), the older Google TS linting we had prevented it as well.

Reply:
Okay, adjusting to add a `Creators` suffix.